src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseFlagsServiceTest.java

```diff
@@ -43,4 +54,162 @@ void setupCaseFlags_setsShellCaseFlags() {
         assertThat(respondentFlags.getPartyName(), is(RESPONDENT_NAME));
         assertThat(respondentFlags.getRoleOnCase(), is("respondent"));
     }
+
+    @Test
+    void setDefaultFlags_setsDefaultFlags() {
+        caseFlagsService.setDefaultFlags(caseData);
+
+        assertThat(caseData.getCaseRestrictedFlag(), is(NO));
+        assertThat(caseData.getAutoListFlag(), is(YES));
+        assertThat(caseData.getCaseInterpreterRequiredFlag(), is(NO));
+        assertThat(caseData.getCaseAdditionalSecurityFlag(), is(NO));
+    }
+
+    @Test
+    void processNewlySetCaseFlags_claimantNeedsLanguageInterpreter_setsInterpreterRequiredFlagTrue() {
+        caseData.setClaimantFlags(CaseFlagsType.builder().details(ListTypeItem.from()).build());
+        caseData.setRespondentFlags(CaseFlagsType.builder().details(ListTypeItem.from()).build());
+
+        caseData.getClaimantFlags().getDetails().add(
+                GenericTypeItem.from(FlagDetailType.builder().name(LANGUAGE_INTERPRETER).status(ACTIVE).build())
+        );
+
+        caseFlagsService.processNewlySetCaseFlags(caseData);
+
+        assertThat(caseData.getCaseInterpreterRequiredFlag(), is(YES));
+    }
+
+    @Test
+    void processNewlySetCaseFlags_respondentNeedsLanguageInterpreter_setsInterpreterRequiredFlagTrue() {
+        caseData.setClaimantFlags(CaseFlagsType.builder().details(ListTypeItem.from()).build());
+        caseData.setRespondentFlags(CaseFlagsType.builder().details(ListTypeItem.from()).build());
+
+        caseData.getRespondentFlags().getDetails().add(
+                GenericTypeItem.from(FlagDetailType.builder().name(LANGUAGE_INTERPRETER).status(ACTIVE).build())
+        );
+
+        caseFlagsService.processNewlySetCaseFlags(caseData);
+
+        assertThat(caseData.getCaseInterpreterRequiredFlag(), is(YES));
+    }
+
+    @Test
+    void processNewlySetCaseFlags_claimantNeedsSignLanguageInterpreter_setsInterpreterRequiredFlagTrue() {
```

Comment:
Can we just have two ParameterizedTest? one for RespondentFlags and one for ClaimantFlags
Seems to be a lot of redundant test code here.

Reply:
No, although its a lot of repeated code, its very clear from the title what exactly the test is testing, if we start abstracting away this information, it becomes less readable. https://mtlynch.io/good-developers-bad-tests/

Reply:
Disagree on it as discussed